cmd/generate/fs.go

```diff
@@ -0,0 +1,82 @@
+package generate
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing/fstest"
+)
+
+type FSOption func(file *fstest.MapFile) *fstest.MapFile
+
+func PrependFilesWith(input string) FSOption {
+	return func(file *fstest.MapFile) *fstest.MapFile {
+		return &fstest.MapFile{
+			Data: []byte(fmt.Sprintf("%s%s", input, file.Data)),
+		}
+	}
+}
+
+func fileToFS(filePath string, options ...FSOption) (fs.FS, error) {
+	content, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	mapFile := &fstest.MapFile{Data: content}
+	for _, opt := range options {
+		mapFile = opt(mapFile)
+	}
+
+	return fstest.MapFS{
+		filepath.Base(filePath): mapFile,
+	}, nil
+}
+
+func dirToPrefixedFS(directory string, prefix string, options ...FSOption) (fs.FS, error) {
```

Comment:
Another aside, indeterminate future note - it should absolutely be possible to write a zero-alloc, stateless fs prefixing implementation. At some point that might be nice, as we have silly code like this [elsewhere](https://github.com/grafana/grafana/blob/2554c482eaed570432447abe6dcefcb5aa3f590a/pkg/cuectx/load.go#L69), too